{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metabolomics and Data Analysis (Workshop ESPOL)\n",
    "\n",
    "Instructor: Shuzhao Li, Ph.D (Emory University)\n",
    "August 26-30, 2019.\n",
    "\n",
    "The field of metabolomics is undergoing an explosive growth, due to the enormous potential of metabolomics in understanding biology and disease and in improving human health. This also creates a large gap in training on metabolomics data analysis. This course will provide a practical guide to scientists, engineers and students that employ metabolomics in their work, with an emphasis on the understanding and interpretation of the data.\n",
    "\n",
    "The course will cover major areas of metabolomics data analysis, including study design, data processing, statistical and pathway analysis, integration with other data types and data presentation. Brief introduction will be given to related topics of precision medicine, exposome and big data analytics.\n",
    "\n",
    "This site will be updated during the course.\n",
    "* Because it's easier to download the whole site from GitHub than individual files, consider re-downloading this workshop site each morning to get the newer files.\n",
    "* For Jupyter notebooks, please shut down a notebook (kernel) after you finish with it. Each kernel cost computer resources!\n",
    "\n",
    "## Day 1.\n",
    "1. Workshop overview, aligning goals and expectations\n",
    "2. Overview of metabolomics, current opportunities and challenges\n",
    "3. Preparing software - MzMine 2, and sample data\n",
    "4. Preparing data science tools\n",
    "5. Participant presentation and discussion\n",
    "6. Processing software and the landscape of computational metabolomics\n",
    "7. Hand-on session on MzMine 2 and XCMS Online\n",
    "\n",
    "## Day 2.\n",
    "1. A bioinformatics primer to data science. Using Python and R for data analysis.\n",
    "2. Live session of Jupyter Notebook.\n",
    "3. Statistical analysis - overview\n",
    "4. Hand-on session, statistical analysis using Jupyter Notebook.\n",
    "5. Hand-on session, statistical analysis using MetaboAnalyst\n",
    "6. Participant presentation and discussion\n",
    "7. LC-MS metabolomics quality control\n",
    "8. Data visualization and presentation\n",
    "\n",
    "## Day 3.\n",
    "1. Study case of YFV infection of immune cells\n",
    "2. Metabolomics Databases and searches\n",
    "3. Starting hand-on pathway analysis - Using KEGG and MetaboAnalyst\n",
    "4. Pathway and network analysis - lecture\n",
    "5. Continued hand-on session on pathway and network analysis\n",
    "6. Cytoscape and Metscape\n",
    "\n",
    "## Day 4.\n",
    "1. Application of metabolomics to precision medicine\n",
    "2. Metabolite annotation and reporting\n",
    "3. Review of statistical analysis\n",
    "4. Hand-on session on MWAS and data presentation\n",
    "5. Participant project discussions\n",
    "\n",
    "## Day 5.\n",
    "1. Review of metabolomic workflow\n",
    "2. How metabolomics is applied to exposome and environmental health\n",
    "3. Integration of metabolomics in Systems biology\n",
    "4. Review of pathway and network analysis\n",
    "5. Review of data presentation, visualization methods\n",
    "6. Concluding workshop.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Links used in this course\n",
    "=========================\n",
    "\n",
    "Jupyter notebook:\n",
    "https://jupyter.readthedocs.io/en/latest/content-quickstart.html\n",
    "\n",
    "Anaconda, a software distribution for Python/R data science (Use Python 3 for this workshop):\n",
    "https://www.anaconda.com/distribution/\n",
    "\n",
    "MzMine 2, metabolomics data pre-processing\n",
    "http://mzmine.github.io/download.html\n",
    "\n",
    "MetaboAnalyst: https://www.metaboanalyst.ca\n",
    "\n",
    "XCMS Online: https://xcmsonline.scripps.edu\n",
    "\n",
    "Mummichog server: http://mummichog-2.appspot.com/ (via http://mummichog.org)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Other Resources\n",
    "===============\n",
    "\n",
    "MSconvert, part of proteowizard, for converting mass spectrometry data to open formats:\n",
    "http://proteowizard.sourceforge.net/download.html\n",
    "\n",
    "Metabolic pathways and models\n",
    "\n",
    "KEGG: http://www.genome.jp/kegg/kegg2.html\n",
    "\n",
    "BioCyc: https://biocyc.org\n",
    "\n",
    "Recon: https://www.vmh.life/#human/all\n",
    "\n",
    "Metabolite ID conversion\n",
    "\n",
    "https://cts.fiehnlab.ucdavis.edu\n",
    "\n",
    "https://www.metaboanalyst.ca/faces/upload/ConvertView.xhtml\n",
    "\n",
    "Metabolomics data analysis tools\n",
    "\n",
    "MetScape: http://metscape.ncibi.org/\n",
    "\n",
    "MetExplore: https://metexplore.toulouse.inra.fr/\n",
    "\n",
    "Metabox/Met-DA: http://metda.fiehnlab.ucdavis.edu\n",
    "\n",
    "Network visualization tools\n",
    "\n",
    "Cytoscape: http://cytoscape.org\n",
    "\n",
    "Gehpi: https://gephi.org\n",
    "\n",
    "The UAB Metabolomics Workshops: https://www.uab.edu/proteomics/metabolomics/workshop/workshop_july_2018.php"
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "-all",
   "main_language": "python",
   "notebook_metadata_filter": "-all"
  },
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
