# Metabolomics and Data Analysis (Workshop ESPOL)

Instructor: Shuzhao Li, Ph.D (Emory University)
August 26-30, 2019.

The field of metabolomics is undergoing an explosive growth, due to the enormous potential of metabolomics in understanding biology and disease and in improving human health. This also creates a large gap in training on metabolomics data analysis. This course will provide a practical guide to scientists, engineers and students that employ metabolomics in their work, with an emphasis on the understanding and interpretation of the data.

The course will cover major areas of metabolomics data analysis, including study design, data processing, statistical and pathway analysis, integration with other data types and data presentation. Brief introduction will be given to related topics of precision medicine, exposome and big data analytics.

This site will be updated during the course.

## Day 1.
    1. Workshop overview, aligning goals and expectations
    2. Overview of experimental methods and study design in metabolomics, and statistical considerations
    3. Software - MzMine 2, and sample data
    4. Preparing data science tools
    5. Participant presentation and discussion
    6. Processing software and the landscape of computational metabolomics
    7. Hand-on session on MzMine 2 and XCMS Online

## Day 2.
    1. A bioinformatics primer to data science. Using Python and R for data analysis.
    2. LC-MS metabolomics quality control
    3. Statistical analysis
    4. Participant presentation and discussion
    5. Hand-on session on using interactive command line and using MetaboAnalyst
    6. Data visualization and presentation

## Day 3.
    1. Application of metabolomics to precision medicine
    2. Participant presentation and discussion
    3. Pathway and network analysis
    4. Hand-on session on pathway and network analysis
    5. Continued discussion on data visualization and presentation

## Day 4.
    1. How metabolomics is applied to exposome and environmental health
    2. Metabolite annotation and reporting
    3. Review of statistical methods and pathway/network tools
    4. Hand-on session on MWAS and data presentation
    5. Participant project discussions

## Day 5.
    1. Integration of metabolomics in Systems biology
    2. Model driven integration
    3. Data driven integration, including Hierarchical community approaches
    4. Participant presentation and discussion
    5. Hand-on session of integrative data analysis
    6. Questions and discussions


Links used in this course
=========================

Jupyter notebook:
https://jupyter.readthedocs.io/en/latest/content-quickstart.html

Anaconda, a software distribution for Python/R data science:
https://www.anaconda.com/distribution/

MzMine 2, metabolomics data pre-processing
http://mzmine.github.io/download.html

MetaboAnalyst: https://www.metaboanalyst.ca

XCMS Online: https://xcmsonline.scripps.edu

Mummichog server: http://mummichog-2.appspot.com/ (via http://mummichog.org)


Other Resources
===============

    Metabolic pathways and models

    KEGG: http://www.genome.jp/kegg/kegg2.html

    BioCyc: https://biocyc.org

    Recon: https://www.vmh.life/#human/all

    Metabolite ID conversion

    https://cts.fiehnlab.ucdavis.edu

    https://www.metaboanalyst.ca/faces/upload/ConvertView.xhtml

    Metabolomics data analysis tools

    MetScape: http://metscape.ncibi.org/

    MetExplore: https://metexplore.toulouse.inra.fr/

    Metabox/Met-DA: http://metda.fiehnlab.ucdavis.edu

    Network visualization tools

    Cytoscape: http://cytoscape.org

    Gehpi: https://gephi.org

    The UAB Metabolomics Workshops: https://www.uab.edu/proteomics/metabolomics/workshop/workshop_july_2018.php
