<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="stylesheet" type="text/css" href="../css/jt.css">
<link rel="stylesheet" type="text/css" href="../css/readable.css">
<link rel="stylesheet" type="text/css" href="../css/toc2.css">
<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />
<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>
<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>
<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=analysisDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=analysisArray;
            var docs_map=analysisArrayMap;
            var pos=analysisArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>
<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>
<title>Metabolomics and Data Analysis (Workshop ESPOL), instructor: Shuzhao Li, Ph.D</title>
<style type = "text/css">
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>
<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">
<!-- code folding -->
<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Metabolomics and Data Analysis (Workshop ESPOL), instructor: Shuzhao Li, Ph.D</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Overview</a>
</li>
<li>
  <a href="../analysis.html">Analysis</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="https://github.com/"> source </a>
</li>
</ul>
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Day-5-schedule">Day 5 schedule<a class="anchor-link" href="#Day-5-schedule">&#182;</a></h1><p>Day 5 focus is reviewing of the workshop materials and systems biology integration.</p>
<p>Questions and discussions are imbedded in each session.</p>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="1.-Project-discussions">1. Project discussions<a class="anchor-link" href="#1.-Project-discussions">&#182;</a></h2><h2 id="2.-Review-of-metabolomic-workflow">2. Review of metabolomic workflow<a class="anchor-link" href="#2.-Review-of-metabolomic-workflow">&#182;</a></h2><ul>
<li><p>Biological question and study design</p>
<ul>
<li>Planning statistical analysis before launching study</li>
<li>Power calculation from models or experience</li>
<li>Choose of platforms. Availability and cost. Targeted or untargeted? GC or LC? LC-MS or LC-MS/MS?</li>
<li>Randomization of samples</li>
</ul>
</li>
<li><p>Metabolite annotation and identification</p>
<ul>
<li>Building in-house chemical reference library - most reliable annotation</li>
<li>Computational annotation is helpful but not golden standard</li>
<li>Reporting metabolite identification should follow community standards, e.g. Schymanski et al, 2014.</li>
<li>Dedicated effort on metabolite identification is performed on needed basis, often for significant features found in a study</li>
</ul>
</li>
<li><p>For global profiling using high-resolution instruments, targeted data can be the annotated portion of untargeted data.</p>
<ul>
<li>Acquisition of data</li>
<li>Data format conversion is often needed. MSconvert (part of ProteoWizard) is used to convert proprietary formats from manufacturers to open formats, such as mzXML or mzML. netCDF is another open format, and can be converted from manufacturer's software such as XCalibre!.</li>
<li>Preprocessing, i.e. peak picking and alignment. Isotopes and adducts are often dealt with at this step. Popular open source software include XCMS, MZmine 2 and OpenMS. This produces a table of metabolite features. For LC-MS data, this contains m/z, retention time and intensity per feature per injection (sample or replicate).</li>
<li>Annotation. Experimental and/or computational.</li>
<li>Statistical analysis. Comparison or regression. See notes from Day 4 and Day 2.</li>
<li>Pathway and network analysis.</li>
<li>Additional confirmation of study findings.</li>
</ul>
</li>
<li><p>For biomarker studies, the performance of the biomarkers needs to be reported, often as AUROC (area under receiver operator characteristics curve, which plots recall and precision).</p>
<ul>
<li>This usually requires a discovery cohort and a test (validation, or replication) cohort, but pilot studies can be performed on a single cohort.</li>
<li>The biomarker candidates are usually generated by statistical analysis. The selection is from discovery cohort.</li>
<li>The predictive model (usually classifiers like Logistic regression, SVM, RF) is built on the discovery cohort.</li>
<li>Cross validation can be performed in the discovery cohort, but does not constitute the need of test cohort.</li>
</ul>
</li>
</ul>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="3.-How-metabolomics-is-applied-to-exposome-and-environmental-health">3. How metabolomics is applied to exposome and environmental health<a class="anchor-link" href="#3.-How-metabolomics-is-applied-to-exposome-and-environmental-health">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><em>15 minutes break</em></p>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="4.-Integration-of-metabolomics-in-Systems-biology">4. Integration of metabolomics in Systems biology<a class="anchor-link" href="#4.-Integration-of-metabolomics-in-Systems-biology">&#182;</a></h2><p>Two lines of approaches are commonly used: model driven and data driven.
Metabolic pathways and
genome scale metabolic models are often used to connect metabolites and enzymes. As enzymes are coded by genes, this connects to proteins and genes.
Many multi-omics integration methods utilize PLS, to accommodate the variance structures from different data types.
We will also review the Hierarchical community network approaches.</p>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><em>Lunch break</em></p>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="5.-Review-of-pathway-and-network-analysis">5. Review of pathway and network analysis<a class="anchor-link" href="#5.-Review-of-pathway-and-network-analysis">&#182;</a></h2><p>Two types of workflows here: annotation based pathway and network mapping and enrichment analysis, mummichog based approach to predict significant pathways/network modules first, followed by metabolite identification.</p>
<p>Networks are also the workhorse of data integration.</p>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="6.-Review-of-Jupyter-notebooks,-data-presentation,-visualization-methods">6. Review of Jupyter notebooks, data presentation, visualization methods<a class="anchor-link" href="#6.-Review-of-Jupyter-notebooks,-data-presentation,-visualization-methods">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="7.-Concluding-workshop.">7. Concluding workshop.<a class="anchor-link" href="#7.-Concluding-workshop.">&#182;</a></h2>
</div>
</div>
</div>
<hr>
https://github.com/cgpu/espol-workshop
<p><small>Exported from <a href="https://github.com//blob/74c1a976c4cc892c85cbe542fd5a94ca4701f051/analysis/day5.ipynb"><code>analysis/day5.ipynb</code></a> committed by cgpu on Thu Aug 27 01:06:00 2020 <a href="https://github.com//commit/74c1a976c4cc892c85cbe542fd5a94ca4701f051">revision 1, 74c1a97</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
