<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="stylesheet" type="text/css" href="../css/jt.css">
<link rel="stylesheet" type="text/css" href="../css/readable.css">
<link rel="stylesheet" type="text/css" href="../css/toc2.css">
<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />
<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>
<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>
<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=analysisDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=analysisArray;
            var docs_map=analysisArrayMap;
            var pos=analysisArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>
<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>
<title>Metabolomics and Data Analysis (Workshop ESPOL), instructor: Shuzhao Li, Ph.D</title>
<style type = "text/css">
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>
<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">
<!-- code folding -->
<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Metabolomics and Data Analysis (Workshop ESPOL), instructor: Shuzhao Li, Ph.D</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Overview</a>
</li>
<li>
  <a href="../analysis.html">Analysis</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="https://github.com/"> source </a>
</li>
</ul>
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Metabolomics-and-Data-Analysis-(Workshop-ESPOL)">Metabolomics and Data Analysis (Workshop ESPOL)<a class="anchor-link" href="#Metabolomics-and-Data-Analysis-(Workshop-ESPOL)">&#182;</a></h1><p>Instructor: Shuzhao Li, Ph.D (Emory University)
August 26-30, 2019.</p>
<p>The field of metabolomics is undergoing an explosive growth, due to the enormous potential of metabolomics in understanding biology and disease and in improving human health. This also creates a large gap in training on metabolomics data analysis. This course will provide a practical guide to scientists, engineers and students that employ metabolomics in their work, with an emphasis on the understanding and interpretation of the data.</p>
<p>The course will cover major areas of metabolomics data analysis, including study design, data processing, statistical and pathway analysis, integration with other data types and data presentation. Brief introduction will be given to related topics of precision medicine, exposome and big data analytics.</p>
<p>This site will be updated during the course.</p>
<ul>
<li>Because it's easier to download the whole site from GitHub than individual files, consider re-downloading this workshop site each morning to get the newer files.</li>
<li>For Jupyter notebooks, please shut down a notebook (kernel) after you finish with it. Each kernel cost computer resources!</li>
</ul>
<h2 id="Day-1.">Day 1.<a class="anchor-link" href="#Day-1.">&#182;</a></h2><ol>
<li>Workshop overview, aligning goals and expectations</li>
<li>Overview of metabolomics, current opportunities and challenges</li>
<li>Preparing software - MzMine 2, and sample data</li>
<li>Preparing data science tools</li>
<li>Participant presentation and discussion</li>
<li>Processing software and the landscape of computational metabolomics</li>
<li>Hand-on session on MzMine 2 and XCMS Online</li>
</ol>
<h2 id="Day-2.">Day 2.<a class="anchor-link" href="#Day-2.">&#182;</a></h2><ol>
<li>A bioinformatics primer to data science. Using Python and R for data analysis.</li>
<li>Live session of Jupyter Notebook.</li>
<li>Statistical analysis - overview</li>
<li>Hand-on session, statistical analysis using Jupyter Notebook.</li>
<li>Hand-on session, statistical analysis using MetaboAnalyst</li>
<li>Participant presentation and discussion</li>
<li>LC-MS metabolomics quality control</li>
<li>Data visualization and presentation</li>
</ol>
<h2 id="Day-3.">Day 3.<a class="anchor-link" href="#Day-3.">&#182;</a></h2><ol>
<li>Study case of YFV infection of immune cells</li>
<li>Metabolomics Databases and searches</li>
<li>Starting hand-on pathway analysis - Using KEGG and MetaboAnalyst</li>
<li>Pathway and network analysis - lecture</li>
<li>Continued hand-on session on pathway and network analysis</li>
<li>Cytoscape and Metscape</li>
</ol>
<h2 id="Day-4.">Day 4.<a class="anchor-link" href="#Day-4.">&#182;</a></h2><ol>
<li>Application of metabolomics to precision medicine</li>
<li>Metabolite annotation and reporting</li>
<li>Review of statistical analysis</li>
<li>Hand-on session on MWAS and data presentation</li>
<li>Participant project discussions</li>
</ol>
<h2 id="Day-5.">Day 5.<a class="anchor-link" href="#Day-5.">&#182;</a></h2><ol>
<li>Review of metabolomic workflow</li>
<li>How metabolomics is applied to exposome and environmental health</li>
<li>Integration of metabolomics in Systems biology</li>
<li>Review of pathway and network analysis</li>
<li>Review of data presentation, visualization methods</li>
<li>Concluding workshop.</li>
</ol>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Links-used-in-this-course">Links used in this course<a class="anchor-link" href="#Links-used-in-this-course">&#182;</a></h1><p>Jupyter notebook:
<a href="https://jupyter.readthedocs.io/en/latest/content-quickstart.html">https://jupyter.readthedocs.io/en/latest/content-quickstart.html</a></p>
<p>Anaconda, a software distribution for Python/R data science (Use Python 3 for this workshop):
<a href="https://www.anaconda.com/distribution/">https://www.anaconda.com/distribution/</a></p>
<p>MzMine 2, metabolomics data pre-processing
<a href="http://mzmine.github.io/download.html">http://mzmine.github.io/download.html</a></p>
<p>MetaboAnalyst: <a href="https://www.metaboanalyst.ca">https://www.metaboanalyst.ca</a></p>
<p>XCMS Online: <a href="https://xcmsonline.scripps.edu">https://xcmsonline.scripps.edu</a></p>
<p>Mummichog server: <a href="http://mummichog-2.appspot.com/">http://mummichog-2.appspot.com/</a> (via <a href="http://mummichog.org">http://mummichog.org</a>)</p>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Other-Resources">Other Resources<a class="anchor-link" href="#Other-Resources">&#182;</a></h1><p>MSconvert, part of proteowizard, for converting mass spectrometry data to open formats:
<a href="http://proteowizard.sourceforge.net/download.html">http://proteowizard.sourceforge.net/download.html</a></p>
<p>Metabolic pathways and models</p>
<p>KEGG: <a href="http://www.genome.jp/kegg/kegg2.html">http://www.genome.jp/kegg/kegg2.html</a></p>
<p>BioCyc: <a href="https://biocyc.org">https://biocyc.org</a></p>
<p>Recon: <a href="https://www.vmh.life/#human/all">https://www.vmh.life/#human/all</a></p>
<p>Metabolite ID conversion</p>
<p><a href="https://cts.fiehnlab.ucdavis.edu">https://cts.fiehnlab.ucdavis.edu</a></p>
<p><a href="https://www.metaboanalyst.ca/faces/upload/ConvertView.xhtml">https://www.metaboanalyst.ca/faces/upload/ConvertView.xhtml</a></p>
<p>Metabolomics data analysis tools</p>
<p>MetScape: <a href="http://metscape.ncibi.org/">http://metscape.ncibi.org/</a></p>
<p>MetExplore: <a href="https://metexplore.toulouse.inra.fr/">https://metexplore.toulouse.inra.fr/</a></p>
<p>Metabox/Met-DA: <a href="http://metda.fiehnlab.ucdavis.edu">http://metda.fiehnlab.ucdavis.edu</a></p>
<p>Network visualization tools</p>
<p>Cytoscape: <a href="http://cytoscape.org">http://cytoscape.org</a></p>
<p>Gehpi: <a href="https://gephi.org">https://gephi.org</a></p>
<p>The UAB Metabolomics Workshops: <a href="https://www.uab.edu/proteomics/metabolomics/workshop/workshop_july_2018.php">https://www.uab.edu/proteomics/metabolomics/workshop/workshop_july_2018.php</a></p>
</div>
</div>
</div>
<hr>
https://github.com/cgpu/espol-workshop
<p><small>Exported from <a href="https://github.com//blob/74c1a976c4cc892c85cbe542fd5a94ca4701f051/analysis/README.ipynb"><code>analysis/README.ipynb</code></a> committed by cgpu on Thu Aug 27 01:06:00 2020 <a href="https://github.com//commit/74c1a976c4cc892c85cbe542fd5a94ca4701f051">revision 1, 74c1a97</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
