{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Day 5 schedule\n",
    "\n",
    "Day 5 focus is reviewing of the workshop materials and systems biology integration.\n",
    "\n",
    "Questions and discussions are imbedded in each session."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Project discussions\n",
    "\n",
    "## 2. Review of metabolomic workflow\n",
    "\n",
    "- Biological question and study design\n",
    "  - Planning statistical analysis before launching study\n",
    "  - Power calculation from models or experience\n",
    "  - Choose of platforms. Availability and cost. Targeted or untargeted? GC or LC? LC-MS or LC-MS/MS?\n",
    "  - Randomization of samples\n",
    "\n",
    "- Metabolite annotation and identification\n",
    "  - Building in-house chemical reference library - most reliable annotation\n",
    "  - Computational annotation is helpful but not golden standard\n",
    "  - Reporting metabolite identification should follow community standards, e.g. Schymanski et al, 2014.\n",
    "  - Dedicated effort on metabolite identification is performed on needed basis, often for significant features found in a study\n",
    "\n",
    "- For global profiling using high-resolution instruments, targeted data can be the annotated portion of untargeted data.\n",
    "  - Acquisition of data\n",
    "  - Data format conversion is often needed. MSconvert (part of ProteoWizard) is used to convert proprietary formats from manufacturers to open formats, such as mzXML or mzML. netCDF is another open format, and can be converted from manufacturer's software such as XCalibre!.\n",
    "  - Preprocessing, i.e. peak picking and alignment. Isotopes and adducts are often dealt with at this step. Popular open source software include XCMS, MZmine 2 and OpenMS. This produces a table of metabolite features. For LC-MS data, this contains m/z, retention time and intensity per feature per injection (sample or replicate).\n",
    "  - Annotation. Experimental and/or computational.\n",
    "  - Statistical analysis. Comparison or regression. See notes from Day 4 and Day 2.\n",
    "  - Pathway and network analysis.\n",
    "  - Additional confirmation of study findings.\n",
    "\n",
    "- For biomarker studies, the performance of the biomarkers needs to be reported, often as AUROC (area under receiver operator characteristics curve, which plots recall and precision).\n",
    "  - This usually requires a discovery cohort and a test (validation, or replication) cohort, but pilot studies can be performed on a single cohort.\n",
    "  - The biomarker candidates are usually generated by statistical analysis. The selection is from discovery cohort.\n",
    "  - The predictive model (usually classifiers like Logistic regression, SVM, RF) is built on the discovery cohort.\n",
    "  - Cross validation can be performed in the discovery cohort, but does not constitute the need of test cohort."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. How metabolomics is applied to exposome and environmental health"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*15 minutes break*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Integration of metabolomics in Systems biology\n",
    "\n",
    "Two lines of approaches are commonly used: model driven and data driven.\n",
    "Metabolic pathways and\n",
    "genome scale metabolic models are often used to connect metabolites and enzymes. As enzymes are coded by genes, this connects to proteins and genes.\n",
    "Many multi-omics integration methods utilize PLS, to accommodate the variance structures from different data types.\n",
    "We will also review the Hierarchical community network approaches."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Lunch break*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Review of pathway and network analysis\n",
    "\n",
    "Two types of workflows here: annotation based pathway and network mapping and enrichment analysis, mummichog based approach to predict significant pathways/network modules first, followed by metabolite identification.\n",
    "\n",
    "Networks are also the workhorse of data integration."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Review of Jupyter notebooks, data presentation, visualization methods"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. Concluding workshop."
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "-all",
   "main_language": "python",
   "notebook_metadata_filter": "-all"
  },
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
